samples/stock/demo/black-and-white/demo.js

```diff
@@ -0,0 +1,218 @@
+                    units: [
+                        ['minute', [15]]
+                    ]
+                }
+            }, {
+                text: 'M30',
+                title: 'Set timeframe to 30 minutes',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['minute', [30]]
+                    ]
+                }
+            }, {
+                text: 'H1',
+                title: 'Set timeframe to 1 hour',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['hour', [1]]
+                    ]
+                }
+            }, {
+                text: 'H4',
+                title: 'Set timeframe to 4 hours',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['hour', [4]]
+                    ]
+                }
+            }, {
+                text: 'D1',
+                title: 'Set timeframe to 1 day',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['day', [1]]
+                    ]
+                }
+            }, {
+                text: 'W1',
+                title: 'Set timeframe to 1 week',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['week', [1]]
+                    ]
+                }
+            }, {
+                text: 'MN',
+                title: 'Set timeframe to 1 month',
+                dataGrouping: {
+                    forced: true,
+                    units: [
+                        ['month', [1]]
+                    ]
+                }
+            }],
+            selected: 4,
+            inputEnabled: false
+        },
+        series: [{
+            name: 'AAPL',
+            type: 'candlestick',
+            data: data,
+            color: '#000000',
+            lastPrice: {
+                enabled: true,
+                color: '#c0c0c0',
+                label: {
+                    enabled: true,
+                    backgroundColor: '#000000',
+                    padding: 2,
+                    shape: 'rect',
+                    borderRadius: 0
+                }
+            }
+        }],
+        tooltip: {
+            positioner: function () {
+                return {
+                    x: 10,
+                    y: 0
+                };
+            },
+            format: '{series.name} {point.open:.2f} {point.high:.2f} {point.low:.2f} {point.close:.2f}'
+        },
+        xAxis: {
+            crosshair: {
+                snap: false,
+                label: {
+                    enabled: true,
+                    format: '{value:%d %b %k:%M}'
+                }
+            },
+            startOnTick: true,
+            labels: {
+                distance: 4,
+                formatter() {
```

Comment:
Did you consider using [dateTimeLabelFormats](https://api.highcharts.com/highcharts/xAxis.dateTimeLabelFormats)? It should be able to cover the case with `isAboveD1`, but probably not the `isFirst` case. You might be able to hack `isFirst` through a format string though.

Reply:
I didn't consider it to be honest due to the `isFirst` check. But I guess, we would prefer that approach instead of callbacks, right?

Could you point me in a right direction with this `isFirst` hack? I don't see how I could do it honestly.

Reply:
Looking closer I don't think it's easy, given that we're checking the current data grouping. We'd end up with custom code anyway, so let's just keep the formatter.

Reply:
Cool, thanks!